lib/matplotlib/tests/test_datetime.py

```diff
@@ -433,11 +433,18 @@ def test_fill_betweenx(self):
         ax2.fill_betweenx(y_dates, x_values1, x_values2)
         ax3.fill_betweenx(y_dates, x_dates1, x_dates2)
 
-    @pytest.mark.xfail(reason="Test for hexbin not written yet")
     @mpl.style.context("default")
     def test_hexbin(self):
-        fig, ax = plt.subplots()
-        ax.hexbin(...)
+        np.random.seed(19680801)
+        dates = np.arange(np.datetime64('2022-01-01'), np.datetime64('2023-12-31'),
+                          np.timedelta64(1, 'D'))
+        fig, (ax0, ax1, ax2) = plt.subplots(3, 1, constrained_layout=True)
+
+        n = dates.shape
+        y = np.random.standard_normal(n)
+        ax0.hexbin(dates, y, gridsize=10)
+        ax1.hexbin(dates, y, gridsize=10)
+        ax2.hexbin(dates, y, gridsize=10)
```

Comment:
```suggestion
        ax1.hexbin(y, dates, gridsize=10)
        ax2.hexbin(dates, dates, gridsize=10)
```

Reply:
Didn't commit the most recent changes.  Now there are tweaks to the appearance of the plots and the appropriate hexbin arguments

Reply:
Added the .rcParams["date.converter"] = 'concise' to clean up the datetime labels and updated the image in the body of the PR request.